int8 page_size
int32 starting_log_num
